src/manifest/__tests__/filter_supported_representations.test.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describe("Manifest - filterSupportedRepresentations", () => {
+  beforeEach(() => {
+    jest.resetModules();
+  });
+
+  it("should not filter text adaptations", () => {
+    const isCodecSupportedSpy = jest.fn(() => false);
+    jest.mock("../../compat", () => ({ isCodecSupported: isCodecSupportedSpy }));
+
+    const filterSupportedRepresentations = require("../filter_supported_representations")
+      .default;
+
+    const rep1 = { id: "54", bitrate: 100 };
+    const rep2 = { id: "55", bitrate: 101 };
+    expect(filterSupportedRepresentations("text", [rep1, rep2]))
+      .toEqual([rep1, rep2]);
+
+    expect(isCodecSupportedSpy).not.toHaveBeenCalled();
+  });
+
+  it("should not filter image adaptations", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 22. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=439&issues=AWgpECqg_MPaBDn0IVTx&open=AWgpECqg_MPaBDn0IVTx)

[ISSUE_KEY]: AWgpECqg_MPaBDn0IVTx